Add Qiskit Docs Guide (#173)
Qiskit/qiskit_sphinx_theme: docs_guide/sphinx_guide/how_to_add_code.rst

.. code-block:: text

    .. jupyter-execute::

        your_code

For example, you can write this:

.. code-block:: text

    .. jupyter-execute::

        from qiskit import QuantumCircuit

        qc = QuantumCircuit(2)
        qc.h(0)
        qc.cx(0,1)
        qc.draw('mpl')

The output would be this cell:

.. jupyter-execute::

    from qiskit import QuantumCircuit

    qc = QuantumCircuit(2)
    qc.h(0)
    qc.cx(0,1)
    qc.draw('mpl')

.. note::

    Even though ``jupyter-execute`` can be used for cells without any visualization, it makes the documentation building process
    more complex, so it is recommended to use :mod:`sphinx.ext.doctest` instead. See `this page <https://github.com/Qiskit/qiskit-terra/issues/7661>`_ for more details.

.. note::

    For the ``jupyter-execute`` cells to appear you need to add ``jupyter_sphinx`` to the :confval:`sphinx:extensions` variable in the ``docs/conf.py`` of your repository.

Non-executable cells
--------------------

There are some situations in which executing the code is not convenient, such as when:

> * The code you are showing is not written in Python but other languages like reStructuredText, YAML, Markdown or Rust.

Reviewer: Wording tweak

```suggestion
* The code you are showing is written in languages other than Python, like reStructuredText, YAML, Markdown or Rust.
```
